pkg/compiler/validators/typing.go

```diff
@@ -33,6 +33,20 @@ func (t trivialChecker) CastsFrom(upstreamType *flyte.LiteralType) bool {
 	if isVoid(upstreamType) {
 		return true
 	}
+
+	// If upstream is an enum, it can be consumed as a string downstream
+	if upstreamType.GetEnumType() != nil {
+		if t.literalType.GetSimple() == flyte.SimpleType_STRING {
+			return true
+		}
+	}
+	// If t is an enum, it can be created from a string as Enums as just constrained String aliases
```

Comment:
is this a runtime error then if the string isn't a valid enum value?

Reply:
yup

Reply:
but our entry points will not allow bad values

Reply:
@katrogan thats actually a good point. Can we restrict this in flyteadmin? Where we construct the launchplan?

Reply:
shouldn't the launch plan have to reference the enum type to be valid?